src/screens/Assistant/index.tsx

```diff
@@ -328,12 +328,10 @@ const Assistant: React.FC<Props> = ({
   const noResultReasons = computeNoResultReasons(date, from, to);
 
   const onPressed = useCallback(
-    (tripPattern) =>
-      navigation.navigate('TripDetailsModal', {
-        from: from!,
-        to: to!,
-        tripPatternId: tripPattern.id!,
-        tripPattern: tripPattern,
+    (tripPatterns, index) =>
+      navigation.navigate('TripDetailsRoot', {
+        initialTripPatterns: tripPatterns,
```

Comment:
This makes it harder to do deep linking at some point as you would require details view to get passed entire search set. Might be a different way to handle this by having two different types of details showing depending on deep linking.

Reply:
Actually didn't consider that, glad you're pointing that out. Don't want to make changes that may hinder future deep linking. Would it be sufficient to alter the details component to accept a singular trip pattern as an alternative to a set of trip patterns? I'm assuming that the only change (more or less) would be skipping pagination

Reply:
I think taking a `tripPatternId` in addition and having `initialTripPatterns` and `tripPatternIndex` optional. So using `tripPatternId` to get data but prefill with initialTripPatterns if they exist. Also probably not showing pagination when tripPatternIndex is not defined.

Reply:
Sounds very reasonable!